### PR TITLE
Fix #957 - Provide OSGi Metadata in Manifest

### LINKED ISSUE #957: Provide OSGi Metadata in Manifest
To make it possible to use the lib without any extra work inside OSGi it would be good to include OSGi Metadata in the Manifest. Since this is ignored for no OSGi environments it wont disturb other users.

I'll provide a PR for this.

### PR DESCRIPTION
**Pull request by laeubi**
Add the necessary config to generate OSGi-Metainfos

### CODE REVIEW

**Comment by srowen** [comment]
I'm neutral on it but if it is just a file in META-INF it's not a big deal. Is it really just a static file? Then just the jar plugin config is enough. What does the other plugin do?

**Comment by laeubi** [comment]
The BND plugin generates just some Headers for the MANIFEST.MF file, the bnd.bnd file is not included, just carry the config.
The config for the maven-jar-plugin simply instructs it to take this generated header into acount and merge with the default one.

**Comment by srowen** [comment]
I see. Do you know what the headers are? there are more direct ways with built-in plugins to add headers to the manifest, or add files to META-INF.

**Comment by laeubi** [comment]
Of course you can add the header manually, the advantage of BND is that it commutes required values automatically.

**Comment by laeubi** [comment]
For example Version and exported packages might change then it will be updated automatically.

File: core/pom.xml [hunk 1]
```diff
@@ -60,6 +60,33 @@
...
+				</execution>
+		</executions>
+		</plugin>
+
+		<plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-jar-plugin</artifactId>
+			<configuration>
+				<archive>
+					<manifestFile>${project.build.outputDirectory}/META-INF/MANIFEST.MF</manifestFile>
```
Comment (srowen) [inline core/pom.xml]: This should be the default, no?

File: core/pom.xml [hunk 2]
```diff
@@ -60,6 +60,33 @@
...
     </profile>
   </profiles>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>biz.aQute.bnd</groupId>
+        <artifactId>bnd-maven-plugin</artifactId>
+        <version>3.5.0</version>
+
+		<executions>
```
Comment (srowen) [inline core/pom.xml]: This should use 2-space indent consistently.

File: core/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Bundle-SymbolicName: com.google.zxing.core
+Import-Package: !com.google.zxing*
```
Comment (srowen) [inline core/bnd.bnd]: Does it matter that core isn't the only artifact that exports packages in this namespace? (For example this is why the Java 9 module system won't work here).

File: core/pom.xml (hunk 1 above)
Comment (laeubi) [reply core/pom.xml]: In my tests the maven jar plugin ignores it otherwise :-(

File: core/bnd.bnd (hunk 3 above)
Comment (laeubi) [reply core/bnd.bnd]: Not for this use case.

File: core/pom.xml (hunk 2 above)
Comment (laeubi) [reply core/pom.xml]: I replaced tabs by spaces

File: core/bnd.bnd (hunk 3 above)
Comment (laeubi) [reply core/bnd.bnd]: Just wondering does other modules implement classes in the same packages as the core bundle?

Comment (srowen) [reply core/bnd.bnd]: Yes, from years and years ago. `javase` has some classes in this package too.

File: core/pom.xml [hunk 4]
```diff
@@ -67,24 +67,24 @@
...
-		<executions>
-			<execution>
-				<goals>
-					<goal>bnd-process</goal>
-				</goals>
-				</execution>
-		</executions>
-		</plugin>
+        <executions>
+        	<execution>
```
Comment (srowen) [inline core/pom.xml]: This is still using tabs I think, and indented too deeply.

**Pull request merged**